### ISSUE TITLE
status_details should have a sibling enum attribute

### ISSUE THREAD

**Issue by pagbabian-splunk** [ocsf/ocsf-schema#1024]
Currently, we have 4 status related attributes: `status_id`, `status`, `status_code`, `status_details`. not counting `http_status` which is tied to the spec.

`status_details` should have a sibling enum attribute that can be defined within specific classes, with a 'See specific usage` description clause.

The example that exemplified this is in the Authentication class where we suggest via description examples, `status_details` values:
For `status_id` = "Failure"

USER_DOES_NOT_EXIST
INVALID_CREDENTIALS
ACCOUNT_DISABLED
ACCOUNT_LOCKED_OUT
PASSWORD_EXPIRED

For `status_id` = "Success"

LOGOFF_USER_INITIATED
LOGOFF_OTHER

Some of these may be common and could be enum values, especially for the Failure cases.  In fact, we don't have any normalized way to represent some of these common cases.  We would as always still have an `Other` case.

**Comment by rmouritzen-splunk** [comment]
I don't think this is correct. The `status_details` field, as it is used today in the schema, is a place to put the the original status text from the source event.

Aside from this, changing a string field to an enum sibling would be a breaking change... sort of... This would require all existing uses of `status_detail` to add a `status_detail_id` field with a value of `99` (Other).

The intent of this field (based on current uses in the core schema) is as a free-form text field that can additional detail beyond the `status_id` / `status` enum pair.

The `status_details` (array) case is even more of a breaking change... This is currently an array of strings. There's nowhere to put the enum value.

**Comment by pagbabian-splunk** [comment]
> Aside from this, changing a string field to an enum sibling would be a breaking change... sort of... This would require all existing uses of `status_detail` to add a `status_detail_id` field with a value of `99` (Other).

Yes, this is a showstopper at minimum - thanks @rmouritzen-splunk .

This Issue is somewhat old (and maybe stale at this point). Upon further review, if we really need/want an enumerated set of 'reasons' why an activity failed, we would need to introduce a new enum attribute pair, for example `reason_id` and `reason`, but I'm hesitant at this point to clutter up status with too many options.

However, I think I misunderstood the intent of our description which may be at the root of my issue. In our current description examples for `Authentication` we seem to imply that the string values should be capitalized tokens, rather than free form strings. That is because we used actual Windows error tokens for the example. If this was the intent, we might consider having the description say 

"additional details returned by the source event, (whether mapped or natively produced), for example possible details for mapped Windows logon and logoff events are:"

> The `status_details` (array) case is even more of a breaking change... This is currently an array of strings. There's nowhere to put the enum value.

Agree fully on `status_details` array: we just introduced it in v1.4, for three objects and the dictionary description is "The contextual description of the status, status_id value."
